ddf-rpc-api/src/main/java/com/connexta/ddf/platform/direct/PlatformMethods.java

```diff
@@ -0,0 +1,171 @@
+    this.catalogMethods = catalogMethods;
+    this.filterBuilder = filterBuilder;
+
+    Map<String, RpcMethod> builder = new HashMap<>();
+    builder.put(
+        "ddf.platform/getActions",
+        rpc.method(
+            this::getActionsByIds,
+            "Takes either a list of metacard ids or a query in the form of `ddf.catalog/query`"
+                + " and returns all actions available on the metacards. `params` takes: ONLY ONE OF `ids` "
+                + "(value: List(String - valid metacard ids), max ids:"
+                + ACTION_ID_QUERY_LIMIT
+                + ") or `query` (See parameters available on `ddf.catalog/query`).  "));
+    methods = builder;
+  }
+
+  @Override
+  public Map<String, RpcMethod> getMethods() {
+    return methods;
+  }
+
+  private Object getActionsByIds(Map<String, Object> params) {
+    Optional<List<String>> ids =
+        Optional.of(params)
+            .map(p -> p.get("ids"))
+            .filter(List.class::isInstance)
+            .map(idList -> (List<String>) idList);
+    Optional<Map<String, Object>> queryParams =
+        Optional.of(params)
+            .map(p -> p.get("query"))
+            .filter(Map.class::isInstance)
+            .map(paramMap -> (Map<String, Object>) paramMap);
+
+    final boolean exactlyOneParam = ids.isPresent() ^ queryParams.isPresent();
+    if (!exactlyOneParam) {
+      return rpc.error(
+          Error.INVALID_PARAMS,
+          "Exactly one of `ids` or `query` must be present in request (at least one but not both)");
+    }
+
+    QueryRequest queryRequest;
+    if (queryParams.isPresent()) {
+      ImmutablePair<QueryRequest, Error> queryOrError =
+          catalogMethods.getQueryRequest(queryParams.get());
+      if (queryOrError.getRight() != null) {
+        Error err = queryOrError.getRight();
+        return rpc.error(
+            Error.INVALID_PARAMS,
+            "Issue encountered while getting query",
+            mapOf("irritant", queryParams.get(), "path", asList("params", "query"), "cause", err));
+      }
+      queryRequest = queryOrError.getLeft();
+    } else {
+      if (ids.get().size() > ACTION_ID_QUERY_LIMIT) {
+        return rpc.error(
+            Error.INVALID_REQUEST,
+            "No more than " + ACTION_ID_QUERY_LIMIT + " ids may be sent at a single time.",
+            mapOf("path", asList("params", "ids")));
+      }
+      if (ids.get().isEmpty()) {
+        return rpc.error(
+            Error.INVALID_PARAMS,
+            "Must specify at least one metacard id within `ids`",
+            mapOf("path", asList("params", "ids")));
+      }
+      queryRequest = getIdRequest(ids.get());
+    }
+    QueryResponse queryResponse;
+    try {
+      queryResponse = catalogFramework.query(queryRequest);
+    } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+      return rpc.error(Error.INTERNAL_ERROR, e.getMessage());
+    }
+
+    // TODO (RCZ) - Send to action provider
```

Comment:
Ticket?

Reply:
It was just outdated comment, i removed it.